### DISCUSSION TITLE
A623308A behavior

### DISCUSSION THREAD

**Discussion by emtneutrino** [hneemann/Digital#1056] [General]
When I start a simulation with this circuit, the outputs always seem to be random at adress 0.  It seems like random data is written to the first byte of ram as soon as the sim is started.  I can change the address to 1 and then switch back to 0 and the contents are the same random bits when the sim starts.  I set the ram as program memory and loaded a hex file at sim start time with byte 0 set to 0 and byte 1 set to 1.  Here are the contents of "program.hex":
v2.0 raw
0
1
I expect to see the output to be 0 at adress 0 and 1 at adress 1.
Whenever I hit the start sim button, the data at 0 seems to be random each time I hit the button.
If I connect WE to high and start the simulation it does what I expect it to.  It seems connecting a switch to WE is doing something weird.
I think I'm not understanding how this circuit works or what the settings do.
![Screenshot (79)](https://user-images.githubusercontent.com/79285199/202092937-ba1c33d3-8823-4254-a1b6-2a8b150d03f2.png)
![Screenshot (81)](https://user-images.githubusercontent.com/79285199/202092938-949f81fb-eaa5-4a59-bf3c-5ede707610c0.png)
![Screenshot (82)](https://user-images.githubusercontent.com/79285199/202093761-3529d2ca-2f77-400d-b9b8-81ea56d433d3.png)
![Screenshot (83)](https://user-images.githubusercontent.com/79285199/202094288-0e088f3a-5fe9-45e8-b84f-f6c4e6bf0b00.png)

**Comment by hneemann** [comment]
This behavior is not unusual. At the moment of switching on, all signals of the circuit are more or less undefined. Each gate needs a few gate propagation times until it is settled. During this time such things happen. The random value arises because during this phase the RAM is written. Since only inputs are connected to the data lines, the lines are in high-z state, which leads to random values being written.

**Reply by emtneutrino** [reply]
Thanks for the response.  So, is there a way to reliably preload the ram AND prevent random data being written to the ram at simulation start time?

**Reply by hneemann** [reply]
Use the Reset component to keep the /CE line high during the initialization.

**Reply by emtneutrino** [reply]
Thanks for the hint.  I had to invert the output of the reset component so the ram would enable.  It is working the way I was expecting.  Thanks a lot!!!

**Reply by hneemann** [reply]
Never mind! :smile:
